app/src/main/java/ch/admin/bag/dp3t/MainActivity.java

```diff
@@ -125,9 +129,32 @@ private void checkIntentForActions() {
 			if (tracingViewModel.getTracingStatusInterface().wasContactReportedAsExposed()) {
 				gotoReportsFragment();
 			}
+		} else if (intent.getData() != null) {
+			checkValidCovidcodeIntent(intent.getData().toString());
 		}
 	}
 
+	private void checkValidCovidcodeIntent(String url) {
+		String[] fragmentSplit = url.split("#");
+		if (fragmentSplit.length != 2) return;
+		String urlPrefix = fragmentSplit[0];
+		String covidCode = fragmentSplit[1];
+		if (!urlPrefix.equals(BuildConfig.COVIDCODE_URL)) return;
+		if (covidCode.length() != 12) return;
+		startInformFlow(covidCode);
+	}
+
+	private void startInformFlow(String covidCode) {
+		getSupportFragmentManager()
```

Comment:
If a detail fragment is already opened in the app, the link does not work.

Additionally, if we relaunch the app from the task manager history, the "enter-covidcode" flow is restarted. This should be avoided.

Furthermore, we should check if the user is already in the "reported positive" state and ignore the link in this case.